File: urlshorter/urlshorter.rb
require 'net/http'
require 'openssl'
require 'json'

def get_short_url(url)

  uri = URI('https://www.googleapis.com/urlshortener/v1/url')
  https = Net::HTTP.new(uri.host,uri.port)
  https.use_ssl = true
  https.verify_mode = OpenSSL::SSL::VERIFY_NONE
  req = Net::HTTP::Post.new(uri.path)
  req.body = JSON.generate({longUrl: url})
  req["Content-Type"] = "application/json"
  res = https.request(req)
  json = JSON.parse(res.body)
  puts json['id']
end


if $0 == __FILE__
  puts("must asign a url ") && exit(-1) if ARGV.empty?
  get_short_url ARGV.first
end

File: urlshorter/urlshorter.py
#!/usr/bin/python3

import urllib.request as urlrequest
import json
import os
import sys

def get_short_url(url):
    short_url_service = 'https://www.googleapis.com/urlshortener/v1/url'
    long_url = json.dumps({'longUrl': url})

    request =  urlrequest.Request(short_url_service)
    request.add_header('Content-Type','application/json')

    opener = urlrequest.build_opener()
    output = opener.open(request,bytes(long_url,'UTF-8')).read()

    print(json.loads(output.decode())['id'])

if __name__ == '__main__':
    url = sys.argv[len(sys.argv) - 1]
    get_short_url(url)

